config/config.go

```diff
@@ -46,42 +46,55 @@ const Mainnet protocol.NetworkID = "mainnet"
 const GenesisJSONFile = "genesis.json"
 
 // Local holds the per-node-instance configuration settings for the protocol.
+// !!! WARNING !!!
+//
+// These versioned struct tags need to be maintained CAREFULLY and treated
+// like UNIVERSAL CONSTANTS - they should not be modified once committed.
+//
+// New fields may be added to the Local struct, along with a version tag
+// denoting a new version. When doing so, also update the
+// installer/config.json.example and test/testdata/configs/config-v{n}.json
+//
+// !!! WARNING !!!
 type Local struct {
 	// Version tracks the current version of the defaults so we can migrate old -> new
 	// This is specifically important whenever we decide to change the default value
 	// for an existing parameter.
-	Version uint32
+	Version uint32 `version[0]:"0" version[1]:"1" version[2]:"2" version[3]:"3" version[4]:"4" version[5]:"5" version[6]:"6" version[7]:"7"`
 
 	// environmental (may be overridden)
 	// When enabled, stores blocks indefinitally, otherwise, only the most recents blocks
 	// are being kept around. ( the precise number of recent blocks depends on the consensus parameters )
-	Archival bool
+	Archival bool `version[0]:"false"`
 
 	// gossipNode.go
 	// how many peers to propagate to?
-	GossipFanout  int
-	NetAddress    string
-	ReconnectTime time.Duration
+	GossipFanout int    `version[0]:"4"`
+	NetAddress   string `version[0]:""`
+
+	// 1 * time.Minute = 60000000000 ns
+	ReconnectTime time.Duration `version[0]:"60" version[1]:"60000000000"`
+
 	// what we should tell people to connect to
-	PublicAddress string
+	PublicAddress string `version[0]:""`
```

Comment:
Does it make sense to omit the tag if the Go zero value will be filled in?

Reply:
My plan was to ensure all field in the `Local` struct are versioned, and verify that in a unit test. That way, no one would add a new field and "forget" to associate it with a version.

Reply:
Ah, I see. That's a good idea.